src/slackv3/person.py

```
@@ -107,13 +107,23 @@ def _cache_user_info(self):
                         attribute, ""
                     )
 
-                team_res = self._webclient.team_info(team=res["user"]["team_id"])
-                if team_res["ok"]:
-                    self._user_info["domain"] = team_res["team"]["domain"]
-                else:
-                    log.warning(
-                        f"Failed to fetch team information for userid {self._userid}. Slack error {team_res['ok']}"
-                    )
+                team = None
+                # Normal users
+                if res.get("user").get("team_id"):
+                    team = res["user"]["team_id"]
+
+                # Users in a ORG/grid setup do not have a team ID
+                elif res.get("user").get("enterprise_user"):
+                    team = res.get("user").get("enterprise_user").get("enterprise_id")
+
+                if team:
+                    team_res = self._webclient.team_info(team=team)
+                    if team_res["ok"]:
+                        self._user_info["domain"] = team_res["team"]["domain"]
+                    else:
+                        log.warning(
+                            f"Failed to fetch team information for userid {self._userid}. Slack error {team_res['ok']}"
+                        )
```

The proposed change will silently fail if the slack response data structure is not `user.team_id` or `user.enterprise_user`.  This would make fault finding more complicated.  Can you add an `else` clause to raise an error if the data structure doesn't confirm with the codes assumption.

---

I have added an else when its looking for the team_id/enterprise_user in the users profile and logs a warning.